pymysql/connections.py

```diff
@@ -332,7 +335,11 @@ def read_length_encoded_integer(self):
         Length coded numbers can be anywhere from 1 to 9 bytes depending
         on the value of the first byte.
         """
-        c = self.read_uint8()
+        try:
+            c = self.read_uint8()
+        except IndexError:
+            raise _NoMoreColumnsInRow()
+
```

Comment:
I understand your concerns, still `try` ~ `except` is only a performance issue if an exception is raised. As far as I can read the mysql source code, this is only triggered by the `SHOW SLAVE HOSTS` command if a pre-5.6 slave is connected behind a >=5.6.2 master (thus not submitting `set @slave_uuid='UUID'` command before `COM_BINLOG_DUMP`).
I believe this is an unfairly uncommon command and the case described is very rare, but I understand your concerns and I will run benchmarks. Do you have an automated setup / reporting test? or just timing select on 1M rows is fine with you?